mock_service/mockservice.py

```diff
@@ -0,0 +1,248 @@
+        if not self._registered:
+            self._read_metadata()
+
+            loader_result = PythonDslLoader().load(self._vdb_metadata)
+            self._mocked_datapoints = loader_result.mocked_datapoints
+            for _, datapoint in self._mocked_datapoints.items():
+                datapoint.value_listener = self._on_datapoint_updated
+            self._behaviors = loader_result.behavior_dict
+
+            self._behavior_executor = BehaviorExecutor(
+                self._mocked_datapoints, self._behaviors, self._pending_event_list
+            )
+            self._register_datapoints()
+            self._subscribe_to_mocked_datapoints()
+            self._feed_initial_values()
+            self._registered = True
+
+    def main_loop(self):
+        """Main execution loop which checks if behaviors shall be executed."""
+        # wait for datapoints to be registered
+        while not self._registered:
+            time.sleep(1)
+
+        try:
+            while True:
+                current_tick_time = time.perf_counter()
+                delta_time: float = current_tick_time - self._last_tick
+                self._behavior_executor.execute(delta_time, self._animators)
+
+                remove_animators: List[Animator] = list()
+                for animator in self._animators:
+                    if not animator.is_done():
+                        animator.tick(delta_time)
+                    else:
+                        remove_animators.append(animator)
+
+                for animator in remove_animators:
+                    self._animators.remove(animator)
+
+                self._last_tick = time.perf_counter()
+                time.sleep(0.1)
+        except Exception as exception:
+            log.exception(exception)
+
+    def _on_datapoint_updated(self, datapoint: MockedDataPoint):
+        """Callback whenever the value of a mocked datapoint changes."""
+        self._set_datapoint(datapoint.path, datapoint.data_type, datapoint.value)
+
+    def _read_metadata(self):
+        """Read metadata from data broker."""
+        request = GetMetadataRequest()
+        reply: GetMetadataReply = self._stub_broker.GetMetadata(
+            request, metadata=self._metadata
+        )
+        for metadata in reply.list:
+            self._vdb_metadata[metadata.name] = metadata
+
+    def _register_datapoints(self):
+        """Register mocked datapoints at data broker."""
+        for datapoint in self._mocked_datapoints.values():
+            if datapoint.is_mocked:
+                metadata = self._vdb_metadata[datapoint.path]
+                self._register(datapoint.path, metadata.data_type, metadata.change_type)
+
+    def _feed_initial_values(self):
+        """Provide initial values of all mocked datapoints to data broker."""
+        for datapoint in self._mocked_datapoints.values():
+            if datapoint.data_type is not None:
+                log.error(datapoint.value)
```

Comment:
This was a debug print which slipped through. Removed it! The print will anyway happen in the _set_datapoint method, hence I want to avoid this additional print